packages/nextra-theme-docs/src/utils/use-is-window-resizing.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useRef, useState } from 'react'
+
+export function useIsWindowResizing() {
+  const [isWindowResizing, setIsWindowResizing] = useState(false)
+  const innerState = useRef(false)
+
+  useEffect(() => {
+    let resizeTimer: ReturnType<typeof setTimeout>
+    const handleResize = () => {
+      if (innerState.current === false) {
+        innerState.current = true
+        setIsWindowResizing(true)
+      }
+
+      clearTimeout(resizeTimer)
+      resizeTimer = setTimeout(() => {
+        innerState.current = false
+        setIsWindowResizing(false)
+      }, 200)
+    }
+    window.addEventListener('resize', handleResize)
+
+    return () => {
+      clearTimeout(resizeTimer)
+      window.removeEventListener('resize', handleResize)
+    }
+  }, [])
+
+  return isWindowResizing
+}
```

Comment:
this hook is very heavy for performance since it updates on every pixel change 

https://github.com/user-attachments/assets/c2c5866b-eded-4a2a-bf05-00564f56bd65

Reply:
I used the [preview page](https://nextra-v2-git-renovate-headlessui-react-2x-lockfile-shud.vercel.app/) from other PRs, and it seems like there's a similar issue when resizing. I'm not sure if it's related to this hook.